app/helpers/organizations_helper.rb

```diff
@@ -10,4 +10,18 @@ def can_current_user_edit_role?(member)
   def available_roles_for_member(member_role)
     (member_role == "viewer") ? Accounts::Membership.allowed_roles : Accounts::Membership.all_roles
   end
+
+  def can_current_user_remove_member?(member)
+    return false if current_user.id == member.id  # Users can't remove themselves
+    return false if current_user_role != Accounts::Membership.roles[:owner]
+
+    member_role = member.role_for(current_organization)
+
+    # If trying to remove an owner, check that there will still be at least one owner left
+    if member_role == Accounts::Membership.roles[:owner]
+      current_organization.members.with_role(Accounts::Membership.roles[:owner]).count > 1
+    else
+      true  # Non-owner members can always be removed by owners
+    end
```

Comment:
Since we're doing a discard and not a destroy, `before_destroy` will probably not work, see if we can use something else from https://github.com/jhawthorn/discard